test/cri-containerd/container_test.go

```diff
@@ -309,6 +310,222 @@ func Test_RunContainer_VirtualDevice_GPU_LCOW(t *testing.T) {
+	if err != nil {
+		t.Skipf("skipping test, failed to find assignable nvidia gpu on host with: %v", err)
+	}
+	if testDeviceInstanceID == "" {
+		t.Skipf("skipping test, host has no assignable nvidia gpu devices")
+	}
+
+	pullRequiredLcowImages(t, []string{imageLcowK8sPause, imageLcowAlpine})
+	client := newTestRuntimeClient(t)
+
+	podctx := context.Background()
+	sandboxRequest := &runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name(),
+				Namespace: testNamespace,
+			},
+			Annotations: map[string]string{
+				"io.microsoft.virtualmachine.lcow.kerneldirectboot":                  "false",
+				"io.microsoft.virtualmachine.computetopology.memory.allowovercommit": "false",
+				"io.microsoft.virtualmachine.lcow.preferredrootfstype":               "initrd",
+				"io.microsoft.virtualmachine.devices.virtualpmem.maximumcount":       "0",
+				"io.microsoft.virtualmachine.lcow.vpcienabled":                       "true",
+				// we believe this is a sufficiently large high MMIO space amount for this test.
+				// if a given gpu device needs more, this test will fail to create the container
+				// and may hang.
+				"io.microsoft.virtualmachine.computetopology.memory.highmmiogapinmb": "64000",
+				"io.microsoft.virtualmachine.lcow.bootfilesrootpath":                 testGPUBootFiles,
+			},
+		},
+		RuntimeHandler: lcowRuntimeHandler,
+	}
+
+	podID := runPodSandbox(t, client, podctx, sandboxRequest)
+	defer removePodSandbox(t, client, podctx, podID)
+	defer stopPodSandbox(t, client, podctx, podID)
+
+	device := &runtime.Device{
+		HostPath: "gpu://" + testDeviceInstanceID,
+	}
+
+	containerRequest := &runtime.CreateContainerRequest{
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: "",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Devices: []*runtime.Device{
+				device,
+			},
+			Linux: &runtime.LinuxContainerConfig{},
+			Annotations: map[string]string{
+				"io.microsoft.container.gpu.capabilities": "utility",
+			},
+		},
+		PodSandboxId:  podID,
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	for i := 0; i < numContainers; i++ {
+
+		name := t.Name() + "-Container-" + fmt.Sprintf("%d", i)
+		containerRequest.Config.Metadata.Name = name
+
+		containerID := createContainer(t, client, ctx, containerRequest)
+		defer removeContainer(t, client, ctx, containerID)
+		startContainer(t, client, ctx, containerID)
+		defer stopContainer(t, client, ctx, containerID)
+
+		cmd := []string{"ls", "/dev/nvidia0"}
+
+		containerExecReq := &runtime.ExecSyncRequest{
+			ContainerId: containerID,
+			Cmd:         cmd,
+			Timeout:     10,
+		}
+		response := execSync(t, client, ctx, containerExecReq)
+		if len(response.Stderr) != 0 {
```

Comment:
Nit but can we do
``go 
if response.ExitCode != 0 {
		t.Fatalf("failed with exit code %d running 'ls /dev/nvidia0': %s", response.ExitCode, string(response.Stderr))
}
```